xdsl/dialects/onnx.py

```diff
@@ -892,18 +893,15 @@ def __init__(
     ):
         super().__init__(
             operands=[matrix_A, matrix_B],
-            result_types=[matrix_A.type],
+            result_types=[self.matrix_Y.type],
```

Comment:
At this point, self has not yet been initialised, so there will be no `self.matrix_Y` the safest bet is for the client to pass the type to the init

Reply:
Maybe you could add a test for this in the ONNX frontend?

Reply:
That it's able to construct a correct matmul given an ONNX graph that contains the operation

Reply:
Done